source/common/upstream/upstream_impl.cc

```diff
@@ -1418,8 +1418,17 @@ void ClusterImplBase::onPreInitComplete() {
 void ClusterImplBase::onInitDone() {
   if (health_checker_ && pending_initialize_health_checks_ == 0) {
     for (auto& host_set : prioritySet().hostSetsPerPriority()) {
-      pending_initialize_health_checks_ += host_set->hosts().size();
+      uint64_t health_check_count = 0;
+      for (auto& host : host_set->hosts()) {
+        if (host->disableActiveHealthCheck()) {
+          continue;
+        }
+        ++health_check_count;
+      }
+      pending_initialize_health_checks_ += health_check_count;
```

Comment:
You could forego creating `health_check_count` and just increment `pending_initialize_health_checks_` directly.

Reply:
Good point, done.